### PR TITLE
miniupnpd: update and fix nftables variant

### PR DESCRIPTION
**Pull request by tiagogaspar8**
Description:

Update the package to a commit that fixes an issue with removing PCP
mappings from nftables.
This also allows us to fix the nftables miniupnpd implementation on
openwrt.
In this new implementation, a table is created at the start of miniupnpd
and it is dedicated to miniupnpd with a priority above the firewall4
table. This allows miniupnpd to go ahead of the drop rules of firewall4
and forward traffic as needed. There was the possibility of adding a
chain inside the firewall4 table, but this would raise an issue where
if firewall4 was reloaded the port forwardings would be lost and
miniupnpd could be out of sync. When miniupnpd is stopped the table is
deleted, taking the port forwardings with it.

Some of this commit is based of msylgj's work, mainly the logic of the
init/hotplug scripts and the makefile build parameters.

@stintel @BKPepe and @msylgj I'd love some feedback on this 😄 
And testing also, might work for me but not others.

Signed-off-by: ZiMing Mo <msylgj@immortalwrt.org>
Signed-off-by: Tiago Gaspar <tiagogaspar8@gmail.com>

Maintainer: N/A
Compile tested: WRT3200ACM
Run tested: WRT3200ACM

### CODE REVIEW

**Comment by stintel** [comment]
Ping @jow-

File: net/miniupnpd/Makefile [hunk 1]
```diff
@@ -8,12 +8,15 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=miniupnpd
-PKG_VERSION:=2.2.3
+PKG_VERSION:=3.0.1
```
Comment (stintel) [inline net/miniupnpd/Makefile:11]: Where did you get this version from?

Comment (tiagogaspar8) [reply net/miniupnpd/Makefile:11]: That was what I used for testing, forgot to change this.
May I propose 2.3.0-1? Since 2.3.0 is the latest tag.

Comment (stintel) [reply net/miniupnpd/Makefile:11]: We usually don't use PKG_VERSION when we build packages from a git hash. I'd just drop it.

Comment (tiagogaspar8) [reply net/miniupnpd/Makefile]: Done!

File: net/miniupnpd/files/miniupnpd.init [hunk 2]
```diff
@@ -171,12 +172,29 @@ upnpd() {
 
 		config_foreach conf_rule_add perm_rule
 
+		if [ "Z$FW" = "Zfw4" ]; then
+			#When using nftables configure miniupnpd to use it's own table and chains
```
Comment (stangri) [inline net/miniupnpd/files/miniupnpd.init:176]: "its own".

Comment (tiagogaspar8) [reply net/miniupnpd/files/miniupnpd.init]: Fixed!

**Pull request merged**